datadog/resource_datadog_synthetics_test_.go

```diff
@@ -803,18 +823,17 @@ func resourceDatadogSyntheticsTestRead(d *schema.ResourceData, meta interface{})
 	authV1 := providerConf.AuthV1
 
 	var syntheticsTest datadogV1.SyntheticsTestDetails
+	var syntheticsAPITest datadogV1.SyntheticsAPITest
 	var err error
 	var httpresp *_nethttp.Response
 
-	if d.Get("type") == "browser" {
-		syntheticsTest, httpresp, err = datadogClientV1.SyntheticsApi.GetBrowserTest(authV1, d.Id()).Execute()
-	} else {
-		syntheticsTest, httpresp, err = datadogClientV1.SyntheticsApi.GetTest(authV1, d.Id()).Execute()
+	// get the generic test to detect if it's an api or browser test
+	syntheticsTest, httpresp, err = datadogClientV1.SyntheticsApi.GetTest(authV1, d.Id()).Execute()
 
-		// re-fetch test if it was actually a browser but we didn't have the info earlier
-		if syntheticsTest.GetType() == "browser" {
-			syntheticsTest, _, err = datadogClientV1.SyntheticsApi.GetBrowserTest(authV1, d.Id()).Execute()
-		}
+	if syntheticsTest.GetType() == datadogV1.SYNTHETICSTESTDETAILSTYPE_BROWSER {
+		syntheticsTest, _, err = datadogClientV1.SyntheticsApi.GetBrowserTest(authV1, d.Id()).Execute()
```

Comment:
Why do we get a `SyntheticsTestDetails` here and not a `SyntheticsBrowserTest` ??

Reply:
The `GetBrowserTest` operation was added before the API/browser objects were split. So it's still the "legacy" object returned by the endpoint. We could change it for a `SyntheticsBrowserTest` but it would create breaking changes.

Reply:
The breaking changes would not be in terraform right ? Only in the go client ?
If that's so, I think we should do it, while we still can (go client beta) to avoid carrying this legacy weirdness in the client

Reply:
Yes we can do that, the provider would have to be fixed after the go client is updated but I can do it 👍

Reply:
Made the PR to update: https://github.com/DataDog/datadog-api-spec/pull/911
I'll update this PR after the go client is updated.